source/administration/update_mattermost.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+##Configuration - please adapt it to your environment
+#path to mattermost directory
+mattermostdir=/opt/mattermost
+#path for backup
+backupdir=/opt
+#temporary path for download
+downloaddir=/tmp
+#Team or Enterprise edition? Set 1 for team edition
+teamedition=0
+#start with plugins? Set 1 for starting with plugins active
+plugins=0
+#########
+
+##set version number from argument
+version=$1
+
+#set variable for echo output
+if [ $teamedition -ne 0 ]; then
+        edition=Team
+        downloadlocation=https://releases.mattermost.com/$version/mattermost-team-$version-linux-amd64.tar.gz
+else
+        edition=Enterprise
+        downloadlocation=https://releases.mattermost.com/$version/mattermost-$version-linux-amd64.tar.gz
+fi
+
+#delete old backup folder
+if [ -d "$backupdir/mattermost-backup" ]; then
+        echo Deleting old mattermost backup
+        rm -rf $backupdir/mattermost-backup
+fi
+
+###Get the file
+echo Downloading Mattermost $edition $version
+wget $downloadlocation -P $downloaddir -q
+
+#Stop if download fails
+if [ $? -ne 0 ]; then
+        echo Mattermost download failed
+        exit 1
+fi
+echo Download successfull
+
+#unpack the tar file
+echo unpacking
+tar -x --transform='s,^[^/]\+,\0-upgrade,' -f $downloaddir/mattermost*.gz -C $downloaddir
+
+#stopping mattermost service
+echo Stopping Mattermost service
+if [ -f /bin/systemctl ];  then
+        servicecommand="systemctl stop mattermost"
+else
+        servicecommand="service mattermost stop"
+fi
+$servicecommand
+
+#check if mattermost is still running
+if pgrep mattermost > /dev/null
+then
+        echo Mattermost still running. Update not possible.
+        echo cleaning
+        rm -rf $downloaddir/mattermost-upgrade/
+        rm -f $downloaddir/mattermost*.gz
+        exit 1
+fi
+
+echo Create backup of mattermost directory
```

Comment:
Does it make sense to add notice/warning to back up the database separately according to own's standard procedures?

Reply:
Could add a confirmation question like „Did you backup your Mattermost Database?“. And of course an override for that question.

Reply:
Sounds good to me, thanks.